mlflow/llama_index/tracer.py

```diff
@@ -184,6 +188,17 @@ def new_span(
 
             token = set_span_in_context(span)
             self._span_id_to_token[span.span_id] = token
+
+            # NB: The tool definition is passed to LLM via kwargs, but it is not set
+            # to the LLM/Chat start event. Therefore, we need to handle it here.
+            if span_type in [SpanType.LLM, SpanType.CHAT_MODEL] and (
+                tools := input_args.get("kwargs", {}).get("tools")
```

Comment:
nit: maybe we can define `tools` separately, this if statement seems a bit too complex for one line.